addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/discovery/CcuDiscoveryParticipant.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.homematic.internal.discovery;
+
+import static org.openhab.binding.homematic.HomematicBindingConstants.THING_TYPE_BRIDGE;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.InetAddress;
+import java.net.MulticastSocket;
+import java.net.SocketTimeoutException;
+import java.util.Collections;
+import java.util.concurrent.Future;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.homematic.internal.discovery.eq3udp.Eq3UdpRequest;
+import org.openhab.binding.homematic.internal.discovery.eq3udp.Eq3UdpResponse;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Discovers Homematic CCU's and adds the results to the inbox.
+ *
+ * @author Gerhard Riegler - Initial contribution
+ */
+@Component(immediate = true, service = DiscoveryService.class)
+public class CcuDiscoveryParticipant extends AbstractDiscoveryService {
+    private final Logger logger = LoggerFactory.getLogger(CcuDiscoveryParticipant.class);
+
+    private static final int RECEIVE_TIMEOUT = 3000;
+    private InetAddress broadcastAddress;
+    private MulticastSocket socket;
+    private Future<?> scanFuture;
+
+    public CcuDiscoveryParticipant() {
+        super(Collections.singleton(THING_TYPE_BRIDGE), 5, true);
+    }
+
+    @Override
+    protected void startScan() {
+        startDiscovery();
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        if (scanFuture == null) {
+            scanFuture = scheduler.submit(this::startDiscovery);
```

Comment:
Sorry, but this is not a background discovery - you are running it only at startup.
A background discovery should find any newly added CCU while openHAB is running.

Reply:
then I misunderstood that, i got it now, will rewrite...